migrations/versions/20181220_populate_measures.py

```diff
@@ -0,0 +1,164 @@
+    internal_reference
+FROM
+    max_measure_version;
+"""
+
+create_subtopic_measure_join_records = """
+INSERT INTO subtopic_measure (subtopic_id, measure_id)
+SELECT DISTINCT
+    subtopic.id, measure.id
+FROM
+    measure_version AS mv_subtopic
+JOIN
+    measure_version AS mv_measure
+ON
+    mv_subtopic.id = mv_measure.parent_id
+JOIN
+    subtopic
+ON
+    subtopic.uri = mv_subtopic.uri
+    AND mv_subtopic.page_type = 'subtopic'
+JOIN
+    measure
+ON
+    measure.uri = mv_measure.uri
+    AND mv_measure.page_type = 'measure'
+WHERE
+    mv_subtopic.page_type = 'subtopic'
+    AND mv_measure.page_type = 'measure'
+"""
+
+update_measure_versions_with_ids = """
+UPDATE
+    measure_version AS mv
+SET
+    measure_id = m.id
+FROM
+    (SELECT m.guid, m.id FROM measure AS m JOIN max_measure_version AS max_mv ON m.guid = max_mv.guid) AS m
+WHERE
+    mv.guid = m.guid
+"""
+
+
+def upgrade():
+    # Using a temporary `guid` column on the `measure` table significantly simplifies updates/inserts and ensuring
+    # data validity during the migration
+    op.add_column("measure", sa.Column("guid", sa.String()))
+
+    op.execute(max_measure_version_temp_table)
+    op.execute(copy_topics)
+    op.execute(copy_subtopics)
+    op.execute(copy_measures)
+    op.execute(create_subtopic_measure_join_records)
+    op.execute(update_measure_versions_with_ids)
+
+    op.drop_column("measure", "guid")
+
+    # Run some basic assertions to test data validity
+    conn = op.get_bind()
+
+    count_topics_in_topic_table = conn.execute("SELECT COUNT(*) FROM topic").scalar()
+    count_topics_in_page_table = conn.execute("SELECT COUNT(*) FROM measure_version WHERE page_type = 'topic'").scalar()
+    assert count_topics_in_topic_table == count_topics_in_page_table, (
+        f"Number of topics in monolith table ({count_topics_in_page_table}) does "
+        f"not match normalised table ({count_topics_in_topic_table})"
+    )
+
+    count_subtopics_in_subtopic_table = conn.execute("SELECT COUNT(*) FROM subtopic").scalar()
+    count_subtopics_in_page_table = conn.execute(
+        "SELECT COUNT(*) FROM measure_version WHERE page_type = 'subtopic'"
+    ).scalar()
+    assert count_subtopics_in_subtopic_table == count_subtopics_in_page_table, (
+        f"Number of subtopics in monolith table ({count_subtopics_in_page_table}) does "
+        f"not match normalised table ({count_subtopics_in_subtopic_table})"
+    )
+
+    count_measures_in_measure_table = conn.execute("SELECT COUNT(*) FROM measure").scalar()
+    count_measures_in_page_table = conn.execute(
+        "SELECT guid, COUNT(DISTINCT(parent_guid)) FROM measure_version "
+        "WHERE page_type = 'measure' GROUP BY guid, parent_guid"
+    ).rowcount
+    assert count_measures_in_measure_table == count_measures_in_page_table, (
+        f"Number of measures in monolith table ({count_measures_in_page_table}) does "
+        f"not match normalised table ({count_measures_in_measure_table})"
+    )
+
```

Comment:
Feels like we're missing an assertion that the number of entries in `subtopic_measure` is the same as the number of measures.  Because we only expect each measure to belong to one subtopic (for now).

Reply:
Added